shotover-proxy/src/server.rs

```diff
@@ -27,6 +27,29 @@ impl<T: Decoder<Item = Messages, Error = anyhow::Error> + Clone + Send> CodecRea
 pub trait CodecWriteHalf: Encoder<Messages, Error = anyhow::Error> + Clone + Send {}
 impl<T: Encoder<Messages, Error = anyhow::Error> + Clone + Send> CodecWriteHalf for T {}
 
+fn perform_custom_handling(messages: Messages, tx_out: &UnboundedSender<Messages>) -> Messages {
+    // this code creates a new Vec and uses an iterator with mapping and filtering to
+    // populate it from the original Messages.message Vec.  Any messages that require special
+    // handling are processed here.  Specifically messages with ReturnToSender details.
+    let result = messages
+        .into_iter()
+        .map(|m| {
+            // if there is a protocol error handle it.  always return the original message
+            // it will be filtered out in the next step.
+            if m.details == MessageDetails::ReturnToSender {
+                debug!("processing ReturnToSender: {:?}", &m);
+                tx_out.send(vec![m]).ok();
```

Comment:
`.ok()` turns the Error into an Option, this tends to be used to stop warnings about unused errors.
But we dont want to ignore this error!
At the very least we should do something like:
```rust
if let Err(err) = tx_out.send(vec![m]) {
    error!("Failed to send return to sender message: {:?}", err);
}
```